### PR TITLE
Column chooser sorting and searching

### PR DESCRIPTION
**Pull request by r-franke**
This PR implements #949 and #975.

I think it still needs a few last touches and discussion. 

I wasn't sure what to do with the styling of the input field. I considered styling it like the main search field, but that takes up a lot of space. So input is appreciated on this.

Second part is the sorting of the column names. From Phil's issues I know there is some discussion on this. So maybe we can continue that discussion here and choose an approach. 

I also think that the top of the column chooser is taking up quite a lot of real estate due to white space as it is right now. 

Maybe we could replace the select all and deselect row with a checkbox above the other check-boxes and before the new search input field. I believe it's done more often that way and compresses the header a bit.

**Example**

```
[ x ]  [ search input field                                ]
[   ]  AA
[ x ]  RA
..... and so on
```

**Demo**

filter_demo

**Disclaimer**
Still need to fix some flow errors, but wanted to get the discussion done first.

**Out-of-scope-ninja-fixes** 
#982 is fixed.

### CODE REVIEW

**Comment by jameskerr** [comment]
Hey Remy, this is looking great. Here is how you'll fix those flow errors you're encountering:

```js
type LocalState = {
  searchValue: string
}

export default class ColumnChooserMenu extends React.Component<Props, LocalState> {
  state: LocalState = {
    searchValue: ""
  }
  // ...
}
```

As for styling, there is a nice search input component that can be swapped out with the input element very easily. I also don't think the close button will be necessary for this feature.

```diff
               <div className="search-input">
-                <input
+                <SearchInput
                   id="column-search"
                   type="text"
                   value={this.state.searchValue}
                   onChange={this.handleChange}
                 />
-                <CloseButton
-                  id="column-search-clear"
-                  onClick={this.clearSearch}
-                />
               </div>
```

File: src/css/_column-chooser.scss [hunk 1]
```diff
@@ -63,6 +63,20 @@
     padding-bottom: 12px;
     height: 100%;
   }
+
+  .search-input {
+    width: 100%;
+    align-content: center;
```
Comment (jameskerr) [inline src/css/_column-chooser.scss:69]: I think a `margin-bottom: 12px` looks good here.

**Comment by jameskerr** [comment]
I think it's fine to sort the list alphabetically for now.

File: src/css/_column-chooser.scss (hunk 1 above)
Comment (r-franke) [reply src/css/_column-chooser.scss:69]: Done

**Comment by r-franke** [comment]
New `SearchInput` in use:
image

File: src/js/components/ColumnChooserMenu.js [hunk 2]
```diff
@@ -139,16 +159,39 @@ export default class ColumnChooserMenu extends React.Component<Props> {
...
-            {columns.map((c) => (
-              <ColumnListItem key={`${c.name}-${c.type}`}>
-                <Checkbox
-                  label={c.name}
-                  checked={c.isVisible}
-                  onChange={(e) => this.onColumnClick(e, c)}
+            <ControlListItem>
+              <div className="search-input">
+                <SearchInput
+                  id="column-search"
```
Comment (jameskerr) [inline src/js/components/ColumnChooserMenu.js:165]:
```suggestion
                  autoFocus
```

Looks like the id isn't necessary.

And I think an autoFocus would be helpful here.

File: src/css/_column-chooser.scss [hunk 3]
```diff
@@ -63,6 +63,21 @@
...
+  }
+
+  #column-search {
+    width: 100%;
+  }
+  #column-search-clear {
+    position: absolute;
+    right: -2px;
+    top: -2px;
+  }
```
Comment (jameskerr) [inline src/css/_column-chooser.scss:80]:
```suggestion
```

It doesn't look like these will be necessary anymore.

**Comment by r-franke** [comment]
I think all changes are done, dunno why it still shows the requested changes?

**Pull request merged**